include/cuco/detail/probing_scheme/probing_scheme_impl.inl

```diff
@@ -141,6 +149,20 @@ __host__ __device__ constexpr auto double_hashing<CGSize, Hash1, Hash2>::with_ha
   return double_hashing<cg_size, NewHash1, NewHash2>{hash1, hash2};
 }
 
+template <int32_t CGSize, typename Hash1, typename Hash2>
+template <typename NewHash, typename Enable>
+__host__ __device__ constexpr auto double_hashing<CGSize, Hash1, Hash2>::with_hash_function(
+  NewHash const& hash) const
+{
+  static_assert(cuco::is_tuple_like<NewHash>::value,
+                "The given hasher must be a tuple-like object");
+
+  auto const [hash1, hash2] = cuco::pair{hash};
```

Comment:
Why do we need to cast `hash` to a cuco pair first if we already established that it's a tuple-like object that supports structured bindings?

Reply:
Otherwise, we would do some dispatching like:
```cuda
auto const [hash1, hash2] = [&](){
  if constexpr (is_cuda_std_pair_like<NewHash>) {
    return {cuda::std::get<0>(hash), cuda::std::get<1>(hash)};
  }
  if constexpr (is_std_pair_like<NewHash>::value) {
    return {std::get<0>(hash), std::get<1>(hash)};
  }
}();
```